lib/switchtec.c

```diff
@@ -80,6 +81,11 @@ struct switchtec_dev *switchtec_open(const char *device)
 	char *endptr;
 	struct switchtec_dev *ret;
 
+	if (sscanf(device, "/dev/tty%*[^0-9]%d", &idx) == 1) {
```

Comment:
This is a very poor way to tell if you have been handed a tty.

I'd hook into switchtec_open_by_path() and call isatty() on the fd and split off from there if it is.

Reply:
add isatty check in switchtec_open_uart

Reply:
Not good enough. If I want to specify a device file that doesn't start with "/dev/tty" your code will prevent me from doing so. There's lots of good reasons to do so too, like using socat to forward a TTY over the network or udev rules to symlink the tty device to a more useful name, etc.

Reply:
Yes, it will only process with /dev/tty started.
A question, in socat case or symlink case, will they start with /dev/ for any arbitrary string?

Reply:
No, they can be any path and matching on a path is a bad idea. As I said the first time, hook into switchtec_open_by_path(), once you have an fd call isatty() and if it is a tty call switchtec_open_uart(). Parsing the path in any way is a bad idea.

Reply:
changed, pls comments.